File: src/engine/api/_site.py
"""Single source of truth for site-wide identity strings (backend mirror).

Counterpart of `scandi-desk-main/src/config/site.ts`. The FE constant is
the surface customers see; this module is for backend touchpoints —
HTTPException messages that quote the support email, transactional
email From/Reply-To headers (when an email provider is wired), and the
SALES_INBOX_EMAIL default. Keep the two files in sync — single grep
catches any drift:

    grep -rn "contact@cfo-ai" src/engine scandi-desk-main/src/config

The values can be overridden via env without code edits (useful for
staging where you might want a `staging@` mailbox), defaulting to the
production prod values so a fresh `.env` still produces sane output.
"""
from __future__ import annotations

import os

SITE = {
    # Canonical support inbox. Same address as FE `SITE.supportEmail`.
    # Override via SITE_SUPPORT_EMAIL on staging if needed.
    "support_email": os.environ.get("SITE_SUPPORT_EMAIL", "contact@cfo-ai.io"),
    # Noreply alias — used as From header for one-way transactional
    # emails where replies don't make sense (welcome, password reset).
    # Replies-To still resolves to support_email.
    "noreply_email": os.environ.get("SITE_NOREPLY_EMAIL", "noreply@cfo-ai.io"),
    # Public-facing app name (greeting copy, email subject prefixes).
    "app_name": os.environ.get("SITE_APP_NAME", "CFO AI"),
    # Public domain + URL — mirror FE `SITE.domain` / `SITE.url`.
    "domain": os.environ.get("SITE_DOMAIN", "cfo-ai.io"),
    "url": os.environ.get("SITE_URL", "https://cfo-ai.io"),
}

File: frontend/config/site.ts
// Single source of truth for site-wide URLs + identity strings.
//
// Every component that needs the domain, API host, support email,
// or any other site-identifying string imports from here. NEVER
// hardcode the domain elsewhere — that's how migrations break.
//
// Lint enforcement (eslint.config — see DOMAIN_MIGRATION.md):
//   `no-restricted-syntax` rule against literal "cfo-ai.io" outside
//   this file flags any re-introduction of hardcoded references.
//
// Environment variables shape (Vite — VITE_ prefix is required for
// browser-bundle exposure):
//   VITE_SITE_DOMAIN  — "cfo-ai.io" / "staging.cfo-ai.io" / "localhost:5173"
//   VITE_SITE_URL     — full URL with scheme, no trailing slash
//   VITE_API_URL      — backend base URL (FE fetcher consumes this)
//
// Defaults fall back to production cfo-ai.io so a fresh dev clone with
// no .env still resolves to a working set of URLs (just hitting the
// wrong target — set up local .env to point at localhost).

const env = import.meta.env;

export const SITE = {
  /** Bare hostname (no scheme, no path). Used in meta tags + cookie scoping. */
  domain: (env.VITE_SITE_DOMAIN as string | undefined) ?? "cfo-ai.io",
  /** Full canonical site URL with scheme, no trailing slash. */
  url: (env.VITE_SITE_URL as string | undefined) ?? "https://cfo-ai.io",
  /** Backend API base URL. FE fetcher (src/lib/api.ts, src/lib/rates.ts) consumes this. */
  apiUrl: (env.VITE_API_URL as string | undefined) ?? "https://api.cfo-ai.io",
  /** Product name for headings + email "from" friendly name. */
  appName: "CFO AI",
  /** Legal entity name for footers + Terms / Privacy boilerplate. */
  legalName: "CFO AI",
  /** Contact addresses. `contact@` is the canonical inbox (single mailbox
   *  for support, sales, billing, legal — manually triaged today; if
   *  volume splits we add separate aliases later, but the FE still reads
   *  `supportEmail` so we change one constant and the whole app updates). */
  supportEmail: "contact@cfo-ai.io",
  /** Mailto URL — prefer this in href attributes so we don't sprinkle
   *  `mailto:` literals across the FE. Composes prefilled mailtos via
   *  `${SITE.supportMailto}?subject=...&body=...` at the call site. */
  supportMailto: "mailto:contact@cfo-ai.io",
  noreplyEmail: "noreply@cfo-ai.io",
  /** Social handle (used in Twitter card meta + footer). Update once @cfoai
   *  is verified across platforms; for now keep as placeholder. */
  twitterHandle: "@cfoai",
  /** Documentation / help center root. */
  docsUrl: "https://cfo-ai.io/docs",
} as const;

/** Type for the SITE record — useful in components that read individual
 *  fields and want TS to enforce they're using a real key. */
export type SiteConfig = typeof SITE;
